tools/generator/test/XcodeScheme+ExtensionsTests.swift

```diff
@@ -0,0 +1,396 @@
+            path: .generated("z/A.a")
+        )
+    )
+
+    lazy var libwatchOSx8664TargetID: TargetID = .init("\(libLabel) \(watchOSx8664Configuration)")
+    lazy var libwatchOSx8664Target = Target.mock(
+        label: libLabel,
+        configuration: watchOSx8664Configuration,
+        platform: watchSimulatorPlatform,
+        product: .init(
+            type: .staticLibrary,
+            name: "a",
+            path: .generated("z/A.a")
+        )
+    )
+
+    lazy var libmacOSx8664TargetID: TargetID = .init("\(libLabel) \(macOSx8664Configuration)")
+    lazy var libmacOSx8664Target = Target.mock(
+        label: libLabel,
+        configuration: macOSx8664Configuration,
+        platform: macOSPlatform,
+        product: .init(
+            type: .staticLibrary,
+            name: "a",
+            path: .generated("z/A.a")
+        )
+    )
+
+    lazy var libTestsiOSx8664TargetID: TargetID = .init("\(libTestsLabel) \(applebiniOSiOSx8664Configuration)")
+    lazy var libTestsiOSx8664Target = Target.mock(
+        label: libTestsLabel,
+        configuration: applebiniOSiOSx8664Configuration,
+        platform: iphoneSimulatorPlatform,
+        product: .init(
+            type: .unitTestBundle,
+            name: "a",
+            path: .generated("z/A.a")
+        ),
+        dependencies: [libiOSx8664TargetID]
+    )
+
+    lazy var toolmacOSx8664TargetID: TargetID = .init(
+        "\(toolLabel) \(applebinMacOSDarwinx8664Configuration)")
+    lazy var toolmacOSx8664Target = Target.mock(
+        label: toolLabel,
+        configuration: applebinMacOSDarwinx8664Configuration,
+        platform: macOSPlatform,
+        product: .init(
+            type: .commandLineTool,
+            name: "a",
+            path: .generated("z/A.a")
+        ),
+        dependencies: [libmacOSx8664TargetID]
+    )
+
+    lazy var iOSAppiOSarm64TargetID: TargetID = .init(
+        "\(iOSAppLabel) \(applebiniOSiOSarm64Configuration)")
+    lazy var iOSAppiOSarm64Target = Target.mock(
+        label: iOSAppLabel,
+        configuration: applebiniOSiOSarm64Configuration,
+        platform: iphoneOSPlatform,
+        product: .init(
+            type: .application,
+            name: "a",
+            path: .generated("z/A.a")
+        ),
+        // This target has a dependency on a watchOS app
+        dependencies: [libiOSarm64TargetID, watchOSAppwatchOSarm64TargetID]
+    )
+
+    lazy var iOSAppiOSx8664TargetID: TargetID = .init(
+        "\(iOSAppLabel) \(applebiniOSiOSx8664Configuration)")
+    lazy var iOSAppiOSx8664Target = Target.mock(
+        label: iOSAppLabel,
+        configuration: applebiniOSiOSx8664Configuration,
+        platform: iphoneSimulatorPlatform,
+        product: .init(
+            type: .application,
+            name: "a",
+            path: .generated("z/A.a")
+        ),
+        // This target has a dependency on a watchOS app
+        dependencies: [libiOSx8664TargetID, watchOSAppwatchOSx8664TargetID]
```

Comment:
@brentleyjones I updated the test data such that the iOS app includes a dependency on a watchOS app. Does this approximate the scenario that was mentioned in the other thread?

Reply:
Yes.